### PR TITLE
Avoid nilpointer under certain circumstances

### PR DESCRIPTION
**Pull request by Kidswiss**
## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

File: operator/objectsusercontroller/pipeline.go [hunk 1]
```diff
@@ -45,9 +45,35 @@ func hasSecretRef(ctx *pipelineContext) bool {
...
+	accessKey, exists := csUser.Keys[0]["access_key"]
+	if !exists {
+		accessKey = ""
+	}
+
+	accessSecret, exists := csUser.Keys[0]["secret_key"]
+	if !exists {
+		accessSecret = ""
+	}
+
```
Comment (wejdross) [inline operator/objectsusercontroller/pipeline.go:73]: I don't like this solution, it's ok for avoid nil pointers, but root cause remains the same. If there is a problem with user in OBSERVE, we should trigger UPDATE. With this solution we might end up with working provider, but empty strings in conenction secret

**Review by wejdross — CHANGES_REQUESTED** [review]

File: operator/objectsusercontroller/pipeline.go (hunk 1 above)
Comment (Kidswiss) [reply operator/objectsusercontroller/pipeline.go:73]: The nilpointer didn't happen in observe, it actually happened during the create.

I've checked the connection strings after deploying this for the affected objects, they were still populated as expected.

**Review by wejdross — APPROVED** [review]
LGTM

File: operator/objectsusercontroller/pipeline.go (hunk 1 above)
Comment (wejdross) [reply operator/objectsusercontroller/pipeline.go:73]: Ok thanks for an update, I think this thread can be treated as resolved :)

**Review by TheBigLee — APPROVED** [review]

**Pull request merged**